src/lib.rs

```diff
@@ -224,20 +224,16 @@ where
 {
     let b_len = b.into_iter().count();
 
-    if a.into_iter().next().is_none() {
-        return b_len;
-    }
-
     let mut cache: Vec<usize> = (1..b_len + 1).collect();
 
-    let mut result = 0;
+    let mut result = b_len;
 
     for (i, a_elem) in a.into_iter().enumerate() {
         result = i + 1;
         let mut distance_b = i;
 
         for (j, b_elem) in b.into_iter().enumerate() {
-            let cost = if a_elem == b_elem { 0usize } else { 1usize };
+            let cost = usize::from(a_elem != b_elem);
```

Comment:
I'm surprised that this change makes a difference. Just for my own learning, do you have an explanation for it?

Reply:
In theory this can make a difference because it is a branchless version. This can't lead to CPU misprediction + in this case wouldn't require the branching in the binary. E.g. when implementing a SIMD version of algorithms this would be required, since there is no branching for SIMD.

In praxis this doesn't make a difference, since this is optimized by the compiler already anyways. So this line is simply a cleanup recommended by `cargo clippy`.

Reply:
So the binary size difference is only because we do no longer have to special case empty strings.